### DISCUSSION TITLE
Sharing Components

### DISCUSSION THREAD

**Discussion by purchasingtoolpak** [flyntwp/flynt#377] [Ideas]
Any thoughts to sharing components in a common repository? For example, Kadence will have a cloud to share design templates. 

I just completed my second Flynt theme based website and developed several components. Check them out:
Split Content Overlay: https://thearcofhr.org/our-mission-and-core-values/
Duo Title Box: https://thearcofhr.org/how-we-advocate/
Duo Info Box: https://thearcofhr.org/our-programs/
Block Image Video: https://thearcofhr.org/the-arc-video-gallery/

It would also be great to enhance the Flynt documentation - at a level similar to Timber.

Regards,

Dave

**Comment by steffenbew** [comment]
Hi Dave,

congrats to completing your second Flynt theme!!

I think sharing components would be great. I always thought this might be a huge endeavour. What you had on your mind would be one central GitHub repository people could send pull requests to? Or just a link list with references to others repositories? Sounds much more attainable than a dedicated platform.

Thanks for mentioning you would like improvements to the documentation. Any particulars what you were missing in the documentation when getting started? Did you find the [Getting Started](https://flyntwp.com/the-beginners-guide-to-developing-a-custom-wordpress-theme-with-flynt/) tutorial helpful?

Best
Steffen

**Comment by purchasingtoolpak** [comment]
I thought that the getting started guide was good. 

Could use a guide on the best way to deploy flynt from the desktop to  to the web. Compressing CSS and JS.

A rundown of all of the folders in the theme (what they do).

More information explaining how to change CSS layout, variables and fonts.

More detail on how to create a new component. How to update the functions file to add or change fields. Adding new JS functionality. 

Adding a new WordPress post type.

Using flynt with plugins - e.g. updating the plugin to use the flynt theme header and footer.

**Reply by aaronmeder** [reply]
@purchasingtoolpak this is a great guide on how to deploy Flynt based themes using Github Actions: https://css-tricks.com/continuous-deployments-for-wordpress-using-github-actions/

**Reply by purchasingtoolpak** [reply]
Thanks for sharing this information.

**Comment by timohubois** [comment]
@purchasingtoolpak we are internally thinking about a more elegant way to share custom made components. What would you like to see? And what would you expect from this feature?
